redisenterprise/datadog_checks/redisenterprise/check.py

```diff
@@ -0,0 +1,322 @@
+import sys
+from datetime import datetime, timedelta
+
+import requests
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+# from typing import Any
+
+EVENT_TYPE = SOURCE_TYPE_NAME = 'redisenterprise'
+
+
+class RedisenterpriseCheck(AgentCheck):
+    """RedisenterpriseCheck attempts to connect to the cluster and ensure the node is the master node"""
+
+    def __init__(self, name, init_config, instances):
+        super(RedisenterpriseCheck, self).__init__(name, init_config, instances)
+        # Set this to two minutes ago which may cause duplicates but we need to get everything in the case of failover
+        self.last_event_timestamp_seen = datetime.utcnow() - timedelta(0, 120)
+
+    def _timestamp(self, date):
+        """ Allows us to return an epoch time stamp if we use python2 or python3 """
+        if sys.version_info[0] < 3 or sys.version_info[1] < 4:
+            import time
+
+            return int(time.mktime(date.timetuple()))
+        else:
+            return int(date.timestamp())
+
+    def check(self, instance):
+        host = instance.get('host')
+        port = instance.get('port', 9443)
+        event_limit = instance.get('event_limit', 100)
+        user = instance.get('user')
+        password = instance.get('password')
+        timeout = instance.get('timeout', 10)
+        verifyssl = instance.get('verify_ssl', False)
+        is_mock = instance.get('is_mock', False)
+        service_check_tags = instance.get('tags', [])
```

Comment:
```suggestion
        host = instance.get('host')
        port = instance.get('port', 9443)
        event_limit = instance.get('event_limit', 100)
        user = instance.get('user')
        password = instance.get('password')
        timeout = instance.get('timeout', 10)
        verifyssl = instance.get('verify_ssl', False)
        is_mock = instance.get('is_mock', False)
        service_check_tags = instance.get('tags', [])
```
The usage of the instance parameter in the check method is deprecated. Please use `self.instance` instead

Reply:
@hithwen sorry but the change suggested above is exactly the same.  Is there an example here I can crib off of?

Reply:
I've updated the suggestion